validator/db/kv/attestation_history_v2.go

```diff
@@ -127,6 +127,49 @@ func (hd EncHistoryData) SetTargetData(ctx context.Context, target uint64, histo
 	return hd, nil
 }
 
+// MarkAllAsAttestedSinceLatestWrittenEpoch returns an attesting history with specified target+epoch pairs
+// since the latest written epoch up to the incoming attestation's target epoch as attested for.
+func MarkAllAsAttestedSinceLatestWrittenEpoch(
```

Comment:
Same logic as before, just changed to remove variable shadowing risk https://github.com/prysmaticlabs/prysm/blob/dc897a2007b05cc1aad0f00db13b0c6546c38427/validator/client/attest_protect.go#L135